lib/Sema/CSSimplify.cpp

```diff
@@ -339,6 +339,19 @@ static bool matchCallArgumentsImpl(
     if (numClaimedArgs == numArgs)
       return None;
 
+    // If the next arg is a code completion token with no label, and we're using
+    // the new solver-based code completion (to avoid affecting the AST used by
+    // unmigrated completion kinds) claim it without complaining about the
+    // missing label. Completion will suggest the label in such cases.
+    if (nextArgIdx < numArgs && args[nextArgIdx].getLabel().empty()) {
+      Type nextArgTy = args[nextArgIdx].getPlainType();
+      if (auto *TVT = nextArgTy->getAs<TypeVariableType>()) {
+        if (TVT->getImpl().isCodeCompletionToken() &&
+            TVT->getASTContext().CompletionCallback)
```

Comment:
Maybe remove `CompletionCallback` and wrap whole block (starting at line 346) into `if (cs.isForCodeCompletion())` instead?

Reply:
`matchCallArgumentssImpl` doesn't have `cs` currently. Should I pass it in? I think it'd need to be added to `constraints::matchCallArguments` and `areConservativelyCompatibleArgumentLabels` too.